ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -1,11 +1,134 @@
 
+{-| The logic for how snapshots are created is implicitly spread across different
+  definitions. Here we will describe the general intention on the logic.
+
+  Old-style Snapshots
+  ===================
+
+  The old-style database is an anchored sequence of @ExtLedgerState blk
+  ValuesMK@ which contain a @LedgerState@ which has been deprived from its UTxO
+  table, and said table is stored in the @ValuesMK@ part. These states contain a
+  full UTxO and therefore they are complete pictures of the state of the UTxO
+  after the application of a specific block.
+
+  When we push a ledger state to the old database, we /prune/ it by keeping only
+  the last @k@ elements in the anchored sequence. This means that said list will
+  always have length @<= k + 1@ (plus the anchor). This doesn't imply that the
+  anchor is written to the disk.
+
+  Until the anchor gets serialized into the disk, if the node is shutdown, it
+  will resume operation from an older point. To explain this we will use an
+  example:
+
+  >                                             |<---  k blocks --->|
+  > Selected Chain:  Genesis |> ... > A > B > C  > D > E > F > ... > T
+  > ImmutableDB:     Genesis |> ... > A > B > C
+  > VolatileDB:                               C |> D > E > F > ... > T  (# blocks [D,T] <= k)
+  > Latest Snapshot:                      *
+  > LedgerDB:                                 C |> D > E > F > ... > T
+
+  If the latest snapshot that was taken corresponds to B, if the node is
+  shutdown in this instant, when it restarts, it will load the following state
+  from disk:
+
+  >                                              |<---  k blocks --->|
+  > Selected Chain: Genesis |> ... > A > B >  C  > D > E > F > ... > T
+  > ImmutableDB:    Genesis |> ... > A > B >  C
+  > VolatileDB:                               C |> D > E > F > ... > T  (# blocks [D,T] <= k)
+  > LedgerDB:                            B |>
+
+  Proceeding then to replay the blocks on the ImmutableDB and then the ones on
+  the VolatileDB (the latter will be validated when applied), reaching in the
+  end the same state as above.
+
+  Because each LedgerState is fully contained, there's no problem on the above
+  logic, even if we have some spans of time in which a shutdown will make the
+  node start from an older point.
+
+  It is responsibility of the
+  @Ouroboros.Conensus.Storage.ChainDB.Impl.Background.copyAndSnapshotRunner@
+  copying the blocks to the disk and when dictated by the @DiskPolicy@, take a
+  snapshot. Said snapshot will /always/ be taken from the current anchor of the
+  anchored sequence, as said block is put in the ImmutableDB by the runner
+  before taking the snapshot.
+
+  The @DiskPolicy@ will specify how many snapshots to keep around, as it is
+  possible to start from an older snapshot (due to them being self contained),
+  and also how much time we want at least between snapshots, in order to avoid
+  flushing too quickly.
```

Comment:
Is flushing a concept of the old-style snapshot? Mabe, we should clarify what do we mean by flushing.